internal/xds/balancer/clusterresolver/configbuilder_childname_test.go

```diff
@@ -29,80 +29,420 @@ func Test_nameGenerator_generate(t *testing.T) {
+				nil, // first input is nil
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L1"}}},
+				},
 			},
 			want: []string{"priority-3-0", "priority-3-1"},
 		},
 		{
 			name:   "one new priority",
 			prefix: 1,
-			input1: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}},
-			},
-			input2: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}},
-				{{ID: clients.Locality{Zone: "L1"}}},
+			inputs: [][][]xdsresource.Locality{
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+				},
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L1"}}},
+				},
 			},
 			want: []string{"priority-1-0", "priority-1-1"},
 		},
 		{
 			name:   "merge two priorities",
 			prefix: 4,
-			input1: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}},
-				{{ID: clients.Locality{Zone: "L1"}}},
-				{{ID: clients.Locality{Zone: "L2"}}},
-			},
-			input2: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}, {ID: clients.Locality{Zone: "L1"}}},
-				{{ID: clients.Locality{Zone: "L2"}}},
+			inputs: [][][]xdsresource.Locality{
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L1"}}},
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
+				{
+					{{ID: clients.Locality{Zone: "L0"}}, {ID: clients.Locality{Zone: "L1"}}},
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
 			},
 			want: []string{"priority-4-0", "priority-4-2"},
 		},
 		{
 			name: "swap two priorities",
-			input1: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}},
-				{{ID: clients.Locality{Zone: "L1"}}},
-				{{ID: clients.Locality{Zone: "L2"}}},
-			},
-			input2: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L1"}}},
-				{{ID: clients.Locality{Zone: "L0"}}},
-				{{ID: clients.Locality{Zone: "L2"}}},
+			inputs: [][][]xdsresource.Locality{
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L1"}}},
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
+				{
+					{{ID: clients.Locality{Zone: "L1"}}},
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
 			},
 			want: []string{"priority-0-1", "priority-0-0", "priority-0-2"},
 		},
 		{
 			name: "split priority",
-			input1: [][]xdsresource.Locality{
-				{{ID: clients.Locality{Zone: "L0"}}, {ID: clients.Locality{Zone: "L1"}}},
-				{{ID: clients.Locality{Zone: "L2"}}},
+			inputs: [][][]xdsresource.Locality{
+				{
+					{{ID: clients.Locality{Zone: "L0"}}, {ID: clients.Locality{Zone: "L1"}}},
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
+				{
+					{{ID: clients.Locality{Zone: "L0"}}},
+					{{ID: clients.Locality{Zone: "L1"}}}, // This gets a newly generated name, since "0-0" was already picked.
+					{{ID: clients.Locality{Zone: "L2"}}},
+				},
+			},
+			want: []string{"priority-0-0", "priority-0-1", "priority-0-2"},
```

Comment:
At the end of the first input step, the generated names should be: `[]string{"priority-0-0", "priority-0-1"}`.

When the next step happens, and we are processing the first priority in the list, we will reuse the name `priority-0-0`. When the process the next priority in the list, and see that `L1` was present in the previous config, but its name is already used in the new config, we should ideally generate `priority-0-2` for it, as `priority-0-1` is already used in the previous config. So, the end output should be:
`[]string{"priority-0-0", "priority-0-2", "priority-0-1"},`

Am I missing something?